### PR TITLE
APM-2861 Refactor mock auth to not use selenium

### PR DESCRIPTION
**Pull request by acarriedev**

### CODE REVIEW

File: api_test_utils/oauth_helper.py [hunk 1]
```diff
@@ -72,20 +77,47 @@ async def get_authenticated_with_simulated_auth(self):
...
+                "response_type": "code",
+                "state": "1234567890",
+            },
+        )
 
-        authenticator = _RealAuthFlow(
-            self.base_uri, self.client_id, self.client_secret, self.redirect_uri
+        if resp.status_code != 200:
+            print(json.dumps(resp.json(), indent=2))
+            sys.exit(1)
```
Comment (strutt) [inline api_test_utils/oauth_helper.py:97]: Remove the sys.exit(1) and replace with an exception. It's fine in a script, but we don't want pytest to terminate without doing teardown.

File: api_test_utils/oauth_helper.py [hunk 2]
```diff
@@ -72,20 +77,47 @@ async def get_authenticated_with_simulated_auth(self):
             self.base_uri, self.client_id, self.redirect_uri
         )
         return await authenticator.authenticate()
-
-    async def get_authenticated_with_mock_auth(
-        self, user: str = "9999999999", webdriver_session: WebDriver = None
+
```
Comment (strutt) [inline api_test_utils/oauth_helper.py:80]: Whitespace :-1:

**Review by strutt — CHANGES_REQUESTED** [review]

File: api_test_utils/oauth_helper.py (hunk 1 above)
Comment (acarriedev) [reply api_test_utils/oauth_helper.py]: Replaced with RuntimeError

File: api_test_utils/oauth_helper.py (hunk 2 above)
Comment (acarriedev) [reply api_test_utils/oauth_helper.py]: Removed

**Review by strutt — APPROVED** [review]

**Pull request merged**